workflow/state.go

```diff
@@ -0,0 +1,43 @@
+package workflow
+
+import "github.com/microsoft/durabletask-go/api"
+
+type Status int
+
+const (
+	StatusRunning Status = iota
+	StatusCompleted
+	StatusContinuedAsNew
+	StatusFailed
+	StatusCanceled
+	StatusTerminated
+	StatusPending
+	StatusSuspended
+	StatusUnknown
+)
+
+func (s Status) String() string {
+	status := [...]string{
+		"RUNNING",
+		"COMPLETED",
+		"CONTINUED_AS_NEW",
+		"FAILED",
+		"CANCELED",
+		"TERMINATED",
+		"PENDING",
+		"SUSPENDED",
+	}
+	if s > StatusSuspended || s < StatusRunning {
+		return "UNKNOWN"
+	}
+	return status[s]
+}
+
+type WorkflowState struct {
```

Comment:
Are we using these structs anywhere? It seems they're only referenced here and in `state_test.go`.